banyan/src/error.rs

```diff
@@ -0,0 +1,48 @@
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    #[error("Tree must not be empty")]
+    TreeMustNotBeEmpty,
+
+    #[error("Index out of bounds: {}", .0)]
+    IndexOutOfBounds(u64),
+
+    #[error("must have more than 1 element when extending")]
+    MustHaveMoreThanOneElement,
+
+    #[error("Invalid")]
+    Invalid, // TODO: This error kind sucks
+
+    #[error("Single item too large")]
+    ItemTooLarge,
+
+    #[error("Found purged data")]
+    FoundPurgedData,
+
+    #[error("Full")]
+    Full,
+
+    #[error("Not there")]
+    NotThere,
+
+    #[error("Busy")]
+    Busy,
+
+    #[error("seek offset wraparound")]
```

Comment:
This should probably mention that it applies to the chacha20 cypher stream (perhaps «cypher stream seek offset wrap-around»).